### PR TITLE
feat: enhance getProperty with type casting for returned values

### PR DESCRIPTION
**Pull request by hyyan**
@MatthewHawkins please check the docs , and your demos for any issues

### CODE REVIEW

File: dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java [hunk 1]
```diff
@@ -17,21 +19,48 @@
...
+   * @return the component itself.
+   */
+  public T setProperty(String property, Object value);
+
   /**
    * Retrieves the value of the given property.
    *
+   * @param <V> the type of the property value returned
    * @param property the key of the property
+   * @param typeOfT the type of the property value returned to convert the value to.
```
Comment (MatthewHawkins) [inline dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java:37]: Is this saying that typeOfT is the type that you want the return value to be converted to? If so, that's maybe what I would rephrase this to say (I didn't do it just in case that isn't the meaning).

File: dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java [hunk 2]
```diff
@@ -17,21 +19,48 @@
...
+  public <V> V getProperty(String property, Type typeOfT);
 
   /**
-   * Sets a property value.
+   * Retrieves the value of the given property.
    *
-   * @param property the key of the property
-   * @param value the value to be set
+   * @param <V> the type of the property value returned
+   * @param classOfT the type of the property value returned to convert the value to.
```
Comment (MatthewHawkins) [inline dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java:48]: Same as above

**Review by MatthewHawkins — COMMENTED** [review]
Other than clarification of the two comments in HasProperty, everything looks good!

File: dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java (hunk 1 above)
Comment (hyyan) [reply dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java]: Maybe it should be something like: `the type of the property value returned to cast the value to.` ?!  The method will attempt to convert the value returned by the client to the appropriate Java data type.  However, that this conversion depends on various factors, and although it usually works, there could be cases where it doesn't. So, providing a hint for the data type in such situations is necessary. Can you come with a better explanation
